helm-chart/binderhub/templates/image-cleaner.yaml

```diff
@@ -68,22 +68,33 @@ spec:
       terminationGracePeriodSeconds: 0
       volumes:
       {{- if .Values.imageCleaner.host.enabled }}
-      - name: dockerlib-host
+      - name: storage-host
         hostPath:
           path: {{ .Values.imageCleaner.host.dockerLibDir }}
-      - name: dockersocket-host
+      - name: socket-host
         hostPath:
           path: {{ .Values.imageCleaner.host.dockerSocket }}
           type: Socket
       {{- end }}
-      {{- if .Values.dind.enabled }}
-      - name: dockerlib-dind
+      {{- if eq $Values.imageBuilderType "dind" }}
+      - name: storage-dind
         hostPath:
           path: {{ .Values.dind.hostLibDir }}
           type: DirectoryOrCreate
-      - name: dockersocket-dind
+      - name: socket-dind
         hostPath:
           path: {{ .Values.dind.hostSocketDir }}/docker.sock
           type: Socket
       {{- end }}
+      {{- if eq $Values.imageBuilderType "pink" }}
+      - name: storage-pink
+        hostPath:
+          path: {{ .Values.pink.hostStorageDir }}
+          type: DirectoryOrCreate
+      - name: socket-pink
+        hostPath:
+          path: {{ .Values.pink.hostSocketDir }}/docker.sock
```

Comment:
I have kept `docker.sock` because it is also found in the python code and thought it would keep the coherency. Other than that I don't have strong feelings about the name and we could change it for `podman.sock`.

Reply:
It could be retained, depends on whats involved to change it - but not left undocumented without a comment.

If you add a comment when where this could possibly get someone to think we use a socket for docker when we use actually socket for podman in practice

Reply:
`docker.sock` is hard coded in a lot of places. The podman socket is a docker-compatible interface, so it doesn't matter what it's called. In most cases it gets mapped to `/var/run/docker.sock` inside the container anyway.

For this reason you could also make the case for not having separate `.Values.dind.hostSocketDir` `.Values.pink.hostSocketDir` values on the host system, and that they should both use the same path since they'll never be used together.

I think it's fine as it is, but I'm happy to go with whatever you prefer.

Reply:
I think we can go with the dedicated socket name without too much trouble.

After rechecking the code, I saw that there is a central place that is responsible for handling the hard coded name of the socket in a deployment: the `binderhub_config.py` from the helm chart.

The original logic was doing it and its replacement as well. Therefore, it should be enough to change it there and update the templates to use `podman.sock` when `pink` is selected.

Reply:
If you change it to `podman.sock` you'll also need to modify the image-cleaner to use the new socket name since only the directory is mapped, not the socket file.